libs/vkd3d/swapchain.c

```diff
@@ -317,6 +335,13 @@ static ULONG STDMETHODCALLTYPE dxgi_vk_swap_chain_Release(IDXGIVkSwapChain *ifac
 
     if (!refcount)
     {
+        if (chain->queue->device->vk_info.NV_low_latency2)
+        {
+            pthread_mutex_lock(&chain->present.low_latency_lock);
```

Comment:
This is broken since you're accessing device state here. There needs to be a device-level lock here.

Reply:
I updated this to use the device lock.